server/rooms.ts

```diff
@@ -1935,13 +1933,19 @@ export const Rooms = {
 			} else if (playerOptions && playerOptions.hidden) {
 				privacySetter.add(playerOptions.user.id);
 			}
+			if (playerOptions) battle.checkForcedSettings(playerOptions.user);
+		}
+
+		// Special battles have modchat set to Player from the beginning
+		if (p1Special || battle.forcedSettings.modchat) {
+			room.settings.modchat = '\u2606';
```

Comment:
This can also be moved inside `checkForcedSettings`.

Reply:
Done.